inst/include/population_dynamics/population/population.hpp

```diff
@@ -33,15 +33,457 @@
+        }
+        
+        /**
+         * @brief Calculates total mortality at an index, year, and age
+         * 
+         * @param index_ya dimension folded index for year and age
+         * @param year year index
+         * @param age age index
+         */
+        void CalculateMortality(int index_ya, int year, int age) {
+          for (size_t fleet_=0; fleet_ < this -> nfleets; fleet_++) {
+            int index_yf = year * this -> nfleets + fleet_; //index by fleet and years to dimension fold
+            this -> mortality_F[index_ya] += Fmort[index_yf] * this -> fleets[fleet_] -> selectivity->evaluate(age);
+          }
+          this -> mortality_Z[index_ya] = this -> M[index_ya] + this -> mortality_F[index_ya];
+        }
+
+        
+        /**
+         * @brief Calculates numbers at age at year and age specific indices
+         * 
+         * @param index_ya dimension folded index for year and age
+         * @param index_ya2 dimension folded index for year-1 and age-1
+         */
+        inline void CalculateNumbersAA(int index_ya, int index_ya2) {
+          //using Z from previous age/year - is this correct?
+          this -> numbers_at_age[index_ya] = 
+            this -> numbers_at_age[index_ya2] *
+            (exp(- this -> mortality_Z[index_ya2]));
+        }
+
+        /**
+         * @brief Calculates unfished numbers at age at year and age specific indices
+         * 
+         * @param index_ya dimension folded index for year and age
+         * @param index_ya2 dimension folded index for year-1 and age-1
+         */
+        inline void CalculateUnfishedNumbersAA(int index_ya, int index_ya2) {
+          //using M from previous age/year - is this correct?
+          this -> unfished_numbers_at_age[index_ya] = 
+            this -> unfished_numbers_at_age[index_ya2] *
+            (exp(- this -> M[index_ya2]));
+        }
+        
+        /**
+         * @brief Calculates spawning biomass
+         * 
+         * @param index_ya dimension folded index for year and age
+         * @param year the year spawning biomass is being aggregated for
+         * @param age the age who's biomass is being added into total spawning biomass 
+         */
+        void CalculateSpawningBiomass(int index_ya, int year, int age) {
+          this -> spawning_biomass[year] += this -> proportion_female * 
+            this -> numbers_at_age[index_ya] * 
+            this -> proportion_mature_at_age[age] * 
+            this -> weight_at_age[age];
+        }
+
+        /**
+         * @brief Adds to existing yearly unfished spawning biomass estimates the 
+         *  biomass for a specified year and age
+         * 
+         * @param index_ya dimension folded index for year and age
+         * @param year the year of unfished spawning biomass to add
+         * @param age the age of unfished spawning biomass to add
+         */
+        void CalculateUnfishedSpawningBiomass(int index_ya, int year, int age) {
```

Comment:
I realized we have a mix of how functions work, even when they are all named `Calculate*()`. There are 1) a subset of functions that calculate the value and put it in the object in memory (e.g., `CalculateNumbersAA()`) and then another set of functions that calculate part of the value and ADD it to the value that is in memory (like `CalculateUnfishedSpawningBiomass()`). I think this can be a bit misleading because you aren't actually getting the whole value for the second value upon each call, but for the first set you are getting the whole value.

I wonder if we should refactor to make these all use the same approach (more work) or just call the second set something different (less work; maybe something like AddTo* instead of Calculate*?)

Reply:
@k-doering-NOAA, I agree.